### ISSUE TITLE
Lantern panics [mac]

### ISSUE THREAD

**Issue by fffw** [getlantern/lantern#5217]
Seen from logs at https://github.com/getlantern/forum/issues/1104#issue-173178499

Happens on 2.2.5, but needs to verify in the latest version. Per user comment, Lantern 3.1.1 also quit after several minutes, but without panic logs recorded.

```
Aug 25 11:07:34.217 - 3m31s DEBUG chained: dialer.go:50 Sending CONNECT REQUEST
Jan 01 00:01:28.815 - -24535382m-34s DEBUG balancer: balancer.go:89 Dialing connect://video-sjc2-1.xx.fbcdn.net:443 with (trusted) chained proxy at 162.243.152.193:8174 [obfs4]
Aug 25 11:07:34.536 - 3m31s DEBUG flashlight.util: http.go:279 Creating new HTTPClient with proxyAddrFN: 0x4164140
Aug 25 11:07:34.536 - 3m31s DEBUG flashlight.util: http.go:294 Waiting for proxy server to come online
Aug 25 11:07:34.536 - 3m31s DEBUG flashlight.util: http.go:303 Connected to proxy
Aug 25 11:07:34.536 - 3m31s DEBUG flashlight.logging: logging.go:177 Sending error logs to Loggly via proxy
Aug 25 11:07:34.536 - 3m31s ERROR flashlight: main.go:68 panic: sync: inconsistent mutex state [recovered]
	panic: sync: inconsistent mutex state

goroutine 6572 [running]:
panic(0x44a0200, 0xc8217c4920)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
sync.(*Mutex).Lock(0xc820a20188)
	/usr/local/go/src/sync/mutex.go:75 +0x16d
sync.(*Pool).getSlow(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:130 +0xe8
sync.(*Pool).Get(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:118 +0x141
fmt.newPrinter(0xc82032b800)
	/usr/local/go/src/fmt/print.go:133 +0x27
fmt.Sprintf(0x46d70a0, 0x1d, 0xc820d3b1a8, 0x3, 0x3, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:202 +0x2b
log.(*Logger).Printf(0xc82034e190, 0x46d70a0, 0x1d, 0xc820d3b1a8, 0x3, 0x3)
	/usr/local/go/src/log/log.go:173 +0x49
net/http.(*Server).logf(0xc82026c100, 0x46d70a0, 0x1d, 0xc820d3b1a8, 0x3, 0x3)
	/usr/local/go/src/net/http/server.go:2159 +0x62
net/http.(*conn).serve.func1(0xc82049f180)
	/usr/local/go/src/net/http/server.go:1390 +0x279
panic(0x44a0200, 0xc8217c4900)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
sync.(*Mutex).Lock(0xc820a20188)
	/usr/local/go/src/sync/mutex.go:75 +0x16d
sync.(*Pool).getSlow(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:130 +0xe8
sync.(*Pool).Get(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:118 +0x141
fmt.newPrinter(0xc820d3b450)
	/usr/local/go/src/fmt/print.go:133 +0x27
fmt.Sprintf(0x4635b10, 0x8, 0xc820d3b4b8, 0x3, 0x3, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:202 +0x2b
github.com/getlantern/golog.(*logger).linePrefix(0xc82011a1e0, 0x4, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:133 +0x309
github.com/getlantern/golog.(*logger).printf(0xc82011a1e0, 0x53401d8, 0xc820274e20, 0x4, 0x462cf88, 0x5, 0x46b0630, 0x17, 0xc821813740, 0x3, ...)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:147 +0x3d
github.com/getlantern/golog.(*logger).Debugf(0xc82011a1e0, 0x46b0630, 0x17, 0xc821813740, 0x3, 0x3)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:161 +0xa1
github.com/getlantern/balancer.(*Balancer).Dial(0xc820948140, 0x4630400, 0x7, 0xc821813687, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/balancer/balancer.go:88 +0x5f6
github.com/getlantern/flashlight/client.(*Client).intercept.func2(0x4634800, 0x3, 0xc821813687, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:111 +0x79
github.com/getlantern/detour.Dialer.func1(0x4634800, 0x3, 0xc821813687, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/detour/detour.go:143 +0x9cd
github.com/getlantern/flashlight/client.(*Client).proxiedDialer.func1(0x4634800, 0x3, 0xc821813687, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/client.go:212 +0x323
github.com/getlantern/flashlight/client.(*Client).intercept(0xc82012c280, 0x5350b80, 0xc8219d0820, 0xc8204bd500)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:113 +0xa89
github.com/getlantern/flashlight/client.(*Client).ServeHTTP(0xc82012c280, 0x5350b80, 0xc8219d0820, 0xc8204bd500)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:29 +0x1b4
net/http.serverHandler.ServeHTTP(0xc82026c100, 0x5350b80, 0xc8219d0820, 0xc8204bd500)
	/usr/local/go/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc82049f180)
	/usr/local/go/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2137 +0x44e
panic: sync: inconsistent mutex state [recovered]
	panic: sync: inconsistent mutex state

goroutine 6710 [running]:
panic(0x44a0200, 0xc8217623a0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
sync.(*Mutex).Lock(0xc820a20188)
	/usr/local/go/src/sync/mutex.go:75 +0x16d
sync.(*Pool).getSlow(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:130 +0xe8
sync.(*Pool).Get(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:118 +0x141
fmt.newPrinter(0xc821280cf8)
	/usr/local/go/src/fmt/print.go:133 +0x27
fmt.Sprintf(0x4635b10, 0x8, 0xc821280d60, 0x3, 0x3, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:202 +0x2b
github.com/getlantern/golog.(*logger).linePrefix(0xc82011a9c0, 0x6, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:133 +0x309
github.com/getlantern/golog.(*logger).print(0xc82011a9c0, 0x53489d8, 0xc820bb20a0, 0x6, 0x462d180, 0x5, 0x44a0200, 0xc821762370)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:137 +0x3d
github.com/getlantern/golog.(*errorWriter).Write(0xc82058e070, 0xc8209d3000, 0xd85, 0x1000, 0xd25, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:247 +0x147
log.(*Logger).Output(0xc82034e190, 0x2, 0xc821a73000, 0xd85, 0x0, 0x0)
	/usr/local/go/src/log/log.go:166 +0x37a
log.(*Logger).Printf(0xc82034e190, 0x46d70a0, 0x1d, 0xc8212810c8, 0x3, 0x3)
	/usr/local/go/src/log/log.go:173 +0x7e
net/http.(*Server).logf(0xc82026c100, 0x46d70a0, 0x1d, 0xc8212810c8, 0x3, 0x3)
	/usr/local/go/src/net/http/server.go:2159 +0x62
net/http.(*conn).serve.func1(0xc821110100)
	/usr/local/go/src/net/http/server.go:1390 +0x279
panic(0x44a0200, 0xc82139de40)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
sync.(*Mutex).Lock(0xc820a20188)
	/usr/local/go/src/sync/mutex.go:75 +0x16d
sync.(*Pool).getSlow(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:130 +0xe8
sync.(*Pool).Get(0x505e0e0, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:118 +0x141
fmt.newPrinter(0x8)
	/usr/local/go/src/fmt/print.go:133 +0x27
fmt.Fprintf(0x53489d8, 0xc820274e00, 0x4635b00, 0xc, 0xc821281408, 0x3, 0x3, 0x405496b, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:187 +0x30
github.com/getlantern/flashlight/logging.timestamped.func1(0x53489d8, 0xc820274e00, 0xc821920376, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/logging/logging.go:157 +0x3d1
github.com/getlantern/wfilter.(*simplePrepender).Write(0xc820274e20, 0xc820258300, 0x8c, 0x100, 0x3, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/wfilter/wfilter.go:85 +0x50
fmt.Fprintf(0x53401d8, 0xc820274e20, 0xc821920340, 0x37, 0xc820f88d80, 0x3, 0x3, 0x46b0630, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:189 +0xb2
github.com/getlantern/golog.(*logger).printf(0xc82011a1e0, 0x53401d8, 0xc820274e20, 0x4, 0x462cf88, 0x5, 0x46b0630, 0x17, 0xc820f88d80, 0x3, ...)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:147 +0x12a
github.com/getlantern/golog.(*logger).Debugf(0xc82011a1e0, 0x46b0630, 0x17, 0xc820f88d80, 0x3, 0x3)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/golog/golog.go:161 +0xa1
github.com/getlantern/balancer.(*Balancer).Dial(0xc820948140, 0x4630400, 0x7, 0xc820f88c37, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/balancer/balancer.go:88 +0x5f6
github.com/getlantern/flashlight/client.(*Client).intercept.func2(0x4634800, 0x3, 0xc820f88c37, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:111 +0x79
github.com/getlantern/detour.Dialer.func1(0x4634800, 0x3, 0xc820f88c37, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/detour/detour.go:143 +0x9cd
github.com/getlantern/flashlight/client.(*Client).proxiedDialer.func1(0x4634800, 0x3, 0xc820f88c37, 0x1d, 0x0, 0x0, 0x0, 0x0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/client.go:212 +0x323
github.com/getlantern/flashlight/client.(*Client).intercept(0xc82012c280, 0x5350b80, 0xc820362820, 0xc8201081c0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:113 +0xa89
github.com/getlantern/flashlight/client.(*Client).ServeHTTP(0xc82012c280, 0x5350b80, 0xc820362820, 0xc8201081c0)
	/Users/afisk/gopath/src/github.com/getlantern/lantern/src/github.com/getlantern/flashlight/client/handler.go:29 +0x1b4
net/http.serverHandler.ServeHTTP(0xc82026c100, 0x5350b80, 0xc820362820, 0xc8201081c0)
	/usr/local/go/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc821110100)
	/usr/local/go/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2137 +0x44e
```

**Comment by fffw** [comment]
The location where panic happens in the latest go version https://golang.org/src/sync/mutex.go#L77

It seems unrelated to application code.

**Comment by myleshorton** [comment]
Do you know what Mac version @fffw?

**Comment by fffw** [comment]
@myleshorton macOS Sierra per user's description, which can be obtained only through Beta program.

**Comment by p3rcya** [comment]
A lot of users have complained the incompatibility with macOS Sierra. @myleshorton In China, a lot of tech users use beta versions and it will be great if we can accommodate them

**Comment by AmZaf** [comment]
Now that macOS Sierra is released, Lantern 2.2.5 is no longer usable.

**Comment by p3rcya** [comment]
新版蓝灯3.2已经支持邮箱登陆和MacOS Sierra，请大家去论坛主页 https://github.com/getlantern/forum 下载。感谢各位的耐心等待！
